### DISCUSSION TITLE
Select components - is there a way to automatically highlight the first option in the menu?

### DISCUSSION THREAD

**Discussion by ercgrat** [mantinedev/mantine#3374] [Q&A]
Hi, just wondering about the intended user behavior for Select components in Mantine. Looking at react-select (https://react-select.com/home), if you search a single-select input, the first option in the list is always highlighted in the search. This is an affordance telling the user that hitting `Enter` will select that first result. In the Mantine selects, you can't select anything with just Enter, you always need to hit the down arrow key first, which is cumbersome.

I think my app's users will be really unhappy if we introduce an extra keystroke to select the first search result when we migrate to the Mantine Select. Why doesn't the Select auto-highlight the first result by default? Is there a way to configure it to do so?

So far I am trying a hack approach that dispatches a down arrow key event whenever the search input changes. This works but has the downside of the background color of the first option constantly flashing because the Mantine code is deselecting or refreshing the result set each time before I re-select it.

**Comment by rtivital** [comment]
We had this behavior before and people reported this as being incorrect.

**Reply by noahsb** [reply]
You might be interested this PR got merged recently: https://github.com/mantinedev/mantine/pull/3102
